lib/util/exec/common.ts

```diff
@@ -1,8 +1,150 @@
-import { exec } from 'child_process';
-import { promisify } from 'util';
+import { ChildProcess, spawn } from 'child_process';
 import type { ExecResult, RawExecOptions } from './types';
 
+// https://man7.org/linux/man-pages/man7/signal.7.html#NAME
+// Non TERM/CORE signals
+// The following is step 3. in https://github.com/renovatebot/renovate/issues/16197#issuecomment-1171423890
+const NONTERM = [
+  'SIGCHLD',
+  'SIGCLD',
+  'SIGCONT',
+  'SIGSTOP',
+  'SIGTSTP',
+  'SIGTTIN',
+  'SIGTTOU',
+  'SIGURG',
+  'SIGWINCH',
+];
+
+function stringify(list: Buffer[], encoding: BufferEncoding): string {
+  return Buffer.concat(list).toString(encoding);
+}
+
+function initStreamListeners(
+  cp: ChildProcess,
+  opts: RawExecOptions & { maxBuffer: number; encoding: BufferEncoding }
+): [Buffer[], Buffer[]] {
+  const stdout: Buffer[] = [];
+  const stderr: Buffer[] = [];
+  let stdoutLen = 0;
+  let stderrLen = 0;
+
+  cp.stdout?.on('data', (chunk: Buffer) => {
+    // process.stdout.write(data.toString());
+    const len = Buffer.byteLength(chunk, opts.encoding);
+    stdoutLen += len;
+    if (stdoutLen > opts.maxBuffer) {
+      cp.emit('error', new Error('stdout maxBuffer exceeded'));
+    } else {
+      stdout.push(chunk);
+    }
+  });
+
+  cp.stderr?.on('data', (chunk: Buffer) => {
+    // process.stderr.write(data.toString());
+    const len = Buffer.byteLength(chunk, opts.encoding);
+    stderrLen += len;
+    if (stderrLen > opts.maxBuffer) {
+      cp.emit('error', new Error('stderr maxBuffer exceeded'));
+    } else {
+      stderr.push(chunk);
+    }
+  });
+  return [stdout, stderr];
+}
+
+export function exec(cmd: string, opts: RawExecOptions): Promise<ExecResult> {
+  return new Promise((resolve, reject) => {
+    const encoding = opts.encoding as BufferEncoding;
+    const maxBuffer = opts.maxBuffer ?? 10 * 1024 * 1024; // Set default max buffer size to 10MB
+    const cp = spawn(cmd, {
+      ...opts,
+      detached: process.platform !== 'win32', // force detached on non WIN platforms
```

Comment:
Best reference I've found for this behaviour: https://github.com/nodejs/node/issues/21825#issuecomment-611328888

Reply:
Thanks, that explains the issues I had with `windowHide` and `detached` not working together 😅.